quickwit/quickwit-metastore/src/tests.rs

```diff
@@ -2142,6 +2142,114 @@ pub mod test_suite {
         }
     }
 
+    pub async fn test_metastore_list_splits_with_indexing_timestamp_and_time_range<
```

Comment:
Nitpiick: I would prefer to do the test on published splits as this is the query that will be sent to the metastore.

Reply:
Ok, but really the state could even be removed as that is tested elsewhere. Here we are just using stage split because that is what creates the splits in the metastore.